lib/Db/MessageMapper.php

```diff
@@ -1256,6 +1257,26 @@ public function findLatestMessages(string $userId, array $mailboxIds, int $limit
 		return $this->findRelatedData($this->findEntities($select), $userId);
 	}
 
+	/**
+	 * @param string $userId
+	 * @param int $mailboxId
+	 * @param int $before UNIX timestamp (seconds)
+	 *
+	 * @return Message[]
+	 */
+	public function findMessagesBefore(string $userId, int $mailboxId, int $before): array {
+		$qb = $this->db->getQueryBuilder();
+
+		$select = $qb->select('*')
+			->from($this->getTableName())
+			->where(
+				$qb->expr()->eq('mailbox_id', $qb->createNamedParameter($mailboxId, IQueryBuilder::PARAM_INT), IQueryBuilder::PARAM_INT),
+				$qb->expr()->lt('sent_at', $qb->createNamedParameter($before, IQueryBuilder::PARAM_INT), IQueryBuilder::PARAM_INT)
+			);
+
+		return $this->findRelatedData($this->findEntities($select), $userId);
```

Comment:
How sure are you about findRelatedData? ;)

Reply:
Hmm, the data is not needed. I just copy pasted from another method :sweat_smile:

The call is gone now.